source/fem-dem/gls_sharp_navier_stokes.cc

```diff
@@ -864,16 +883,36 @@ GLSSharpNavierStokesSolver<dim>::refine_ib()
                       // be bigger than the inside radius of the refinement zone
                       // and smaller than the outside radius of the refinement
                       // zone.
-                      if (particles[p].is_inside_crown(
+                      bool is_inside_crown;
+                      if (minimal_crown_refinement_enabled)
+                        {
+                          // The factor depends on dim, since cell->diameter()
+                          // returns the longest diagonal. The diagonal is
+                          // higher in 3D than in 2D. The chosen factor is equal
+                          // to the length of the smallest cell.
```

Comment:
@AmishgaAlphonius I have an explanation for the factor. Does it make sense?

Reply:
I would say that you want the length of one of the edges of the cell. Since the cell diameter returns you the diagonal length of the cell, you assume a cubic cell and divide by the square root of the dimension to get the length of the edge.